packages/react-scripts/scripts/start.js

```diff
@@ -11,6 +11,11 @@
 
 process.env.NODE_ENV = 'development';
 
+// Load environment variables from .env file. Surpress warnings using silent
+// if this file is missing. dotenv will never modify any environment variables
+// that have already been set.
+require('dotenv').config({silent: true});
```

Comment:
Actually `NODE_PATH` *is* already supported since 0.4.0 (https://github.com/facebookincubator/create-react-app/pull/476), sorry I was wrong about this 😊